CMakeLists.txt

```
@@ -12,23 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.14.6)
 project("traiofbits_extensions")
 
 set(ROOT_TEST_TARGET "trailofbits_extensions_tests")
 add_custom_target("${ROOT_TEST_TARGET}")
 
-set(TOB_EXTENSIONS_ROOT "${CMAKE_CURRENT_SOURCE_DIR}")
+set(TOB_EXTENSIONS_ROOT "${CMAKE_CURRENT_SOURCE_DIR}" CACHE STRING "TOB extensions root directory")
 
-set(ENV{OSQUERY_EXTENSION_GROUP_NAME} "trailofbits_osquery_extensions")
-set(ENV{OSQUERY_EXTENSION_GROUP_VERSION} "1.2")
+set(OSQUERY_EXTENSION_GROUP_NAME "trailofbits_osquery_extensions" CACHE STRING "Overrides osquery extension group version" FORCE)
+set(OSQUERY_EXTENSION_GROUP_VERSION "1.2" CACHE STRING "Overrides osquery extension group version" FORCE)
+
+function(generateOsqueryVersion versionstring)
+  string(REPLACE "." ";" version_list ${versionstring} )
+  list(GET version_list 0 major)
+  set(OSQUERY_BUILD_VERSION_MAJOR ${major})
+  list(GET version_list 1 minor)
+  set(OSQUERY_BUILD_VERSION_MINOR ${minor})
+  configure_file(
+    "${CMAKE_CURRENT_SOURCE_DIR}/Version.h.in"
+    "${TOB_EXTENSIONS_ROOT}/Version.h"
+  )
+endfunction()
 
 function(trailofbitsExtensionsMain)
   message(STATUS "================================")
   message(STATUS "Trail of Bits osquery extensions")
-  message(STATUS "Version: $ENV{OSQUERY_EXTENSION_GROUP_VERSION}")
+  message(STATUS "Version: ${OSQUERY_EXTENSION_GROUP_VERSION}")
   message(STATUS "================================")
 
+  generateOsqueryVersion(${OSQUERY_VERSION_AUTODETECTED})
```

```suggestion
  generateOsqueryVersion(${OSQUERY_VERSION_INTERNAL})
```
The autodetected version is only set if the user is not specifying/overriding the version with `-DOSQUERY_VERSION`, `OSQUERY_VERSION_INTERNAL` is what you want instead.